compiler/runtime.go

```diff
@@ -184,6 +184,9 @@ func (e *interpolatedEvaluator) Eval(ctx context.Context, scope Scope) (values.V
 	if err != nil {
 		return nil, err
 	}
+	if o.IsNull() {
+		return values.NewString("<null>"), nil
+	}
```

Comment:
Could you move this logic inside the Stringify function? That way its not duplicated across both compiler and interpreter?